application-analytics-default/src/test/java/com/xwiki/analytics/internal/MostViewedJsonNormaliserTest.java

```diff
@@ -57,44 +66,111 @@ public class MostViewedJsonNormaliserTest
+    public void normalizeDataWithPartialMatchFilter() throws Exception
+    {
+        readJSONS("/mostViewedPages/normalizeDataWithMultipleFilters.json");
+        setupAnyURL();
+        HashMap<String, String> filters = new HashMap<>();
+        filters.put("nb_hits", "27");
+        filters.put("label", "MostViewedPage?editor=wiki");
+        assertEquals(node.get("ResponseWithFilters"),
+            mostViewedJsonNormaliser.normaliseData(node.get("JsonFilters").toString(), filters));
     }
 
-    private void setupURLs(String url1, String url2) throws Exception
+    @Test
+    public void normalizeDataWithMultipleFilters() throws Exception
     {
-        ExtendedURL extendedURL1 = new ExtendedURL(new URL(url1), null);
-        ExtendedURL extendedURL2 = new ExtendedURL(new URL(url2), null);
+        readJSONS("/mostViewedPages/normalizeDataWithPartialMatchFilter.json");
+        setupAnyURL();
+        HashMap<String, String> filters = new HashMap<>();
+        filters.put("nb_hits", "27");
+        filters.put("nb_hits", "27");
+
+        assertEquals(node.get("ResponseWithFilters"),
+            mostViewedJsonNormaliser.normaliseData(node.get("JsonFilters").toString(), filters));
+    }
 
-        ResourceType resourceType1 = resourceTypeResolver.resolve(extendedURL1, Collections.emptyMap());
-        ResourceType resourceType2 = resourceTypeResolver.resolve(extendedURL2, Collections.emptyMap());
+    /**
+     * Will test the case when the url is invalid.
+     */
+    @Test
+    public void normalizeDataWithMalformedUrl() throws Exception
+    {
+        ReflectionUtils.setFieldValue(this.mostViewedJsonNormaliser, "logger", this.logger);
+        readJSONS("/mostViewedPages/normalizeDataWithMalformedUrl.json");
+        HashMap<String, String> filters = new HashMap<>();
+        assertEquals(node.get("MalformedJSONResponse"),
+            mostViewedJsonNormaliser.normaliseData(node.get("MalformedJSON").toString(), filters));
+        verify(logger).warn("Failed to get resource reference from URL: [{}]. Caused by [{}]",
+            "htttp://localhost:8080/xwiki/bin/view/Analytics/Code/MostViwedPages",
+            "MalformedURLException: unknown protocol: htttp");
+    }
 
-        when(resourceReferenceResolver.resolve(extendedURL1, resourceType1, Collections.emptyMap())).thenReturn(null);
-        when(resourceReferenceResolver.resolve(extendedURL2, resourceType2, Collections.emptyMap())).thenReturn(null);
+    @Test
+    public void jsonWithoutURL() throws IOException
+    {
+        readJSONS("/mostViewedPages/jsonWithoutURL.json");
+        assertEquals(node.get("ArrayWithoutURLResponse"),
+            mostViewedJsonNormaliser.normaliseData(node.get("ArrayWithoutURL").toString(), null));
+    }
+
+    private void setupAnyURL() throws Exception
+    {
+        ResourceType resourceType1 = mock(ResourceType.class);
+        when(resourceTypeResolver.resolve(any(ExtendedURL.class), eq(Collections.emptyMap()))).thenReturn(null);
```

Comment:
Why not have this in a [@Before](https://github.com/xwikisas/application-licensing/blob/d168fb88fc0d121bf95e769ea21c55c00bebe5a6/application-licensing-licensor/application-licensing-licensor-api/src/test/java/com/xwiki/licensing/internal/TrialLicenseGeneratorTest.java#L85) method?